include/nbl/asset/interchange/IAssetLoader.h

```diff
@@ -289,6 +289,42 @@ class IAssetLoader : public virtual core::IReferenceCounted
 	SAssetBundle interm_getAssetInHierarchy(IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel, IAssetLoader::IAssetLoaderOverride* _override);
 	SAssetBundle interm_getAssetInHierarchy(IAssetManager* _mgr, io::IReadFile* _file, const std::string& _supposedFilename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel);
 	SAssetBundle interm_getAssetInHierarchy(IAssetManager* _mgr, const std::string& _filename, const IAssetLoader::SAssetLoadParams& _params, uint32_t _hierarchyLevel);
+	//! ECF_DUPLICATE_* flags are ignored (since there's no point in **restoring** if we're not even looking into the cache) -- only ECF_DONT_CACHE_* are honored
+	//! `Args` ought to be same parameter types as interm_getAssetInHierarchy(...) overloads offer
+	template <typename ...Args>
+	SAssetBundle interm_getRestoredAssetInHierarchy(uint32_t _restoreLevels, Args&&... args)
+	{
+		if (_restoreLevels == 0u)
+			return interm_getAssetInHierarchy(std::forward<Args>(args)...);
+
+		auto any_dummy = [](const SAssetBundle& b) {
+			for (const auto& a : b.getContents())
+				if (a->isADummyObjectForCache())
+					return true;
+		};
+
+		auto bundle = interm_getAssetInHierarchy_find(std::forward<Args>(args)...);
```

Comment:
this is wrong, because if you wanted to get a restored asset but are not sure if it was actually loaded before, then:
- this will use the loader to load any unloaded asset, but will not add it to the cache
- return the uncached bundle if none of the objects are dummy 

So it wont be a drop-in replacement for getAssetInHierarchy

The proper way would be to:
- do a "classical" load according to the flags, so cached, uncached or fully duplicate == no overriding except from the usual by the asset loader overload
- use the loader overload to choose your asset as usual
- if the chosen asset is a dummy or their dependant at any point up to the restore level (will never happen if you used duplicate flags for everything, so no need to check explicitly for that) 
   - do a duplicate load up to restore level, after the restore level use the flags specified
   [while doing the duplicate load, you need to allow for the asset loader override's to make about files and stuff if they're not found, as usual, but let it know you're doing it for a restore from dummy = because right now it cant tell the difference between the first primary load and the special reload]
   - for every asset in the duplicate load bundle and the cached load bundle (O(n^2) if one wishes so) = this part I'd implement in the IAssetLoaderOVerride
      - use the asset loader override to choose what should be restored and restore it
  - return the asset the IAssetLoaderOVerride originally chose (which should now not be a dummy)

Reply:
actually in that set-up we could replace most `SAssetBundle interm_getAssetInHierarchy` calls with `IAsset* getRestoredAssetInHierarchy`

Reply:
> this will use the loader to load any unloaded asset, but will not add it to the cache

not true, my flags modifications make sure that we always look into cache, but caching or not caching is still user's decision. Then if assets were just loaded, they're instantly returned (w/o reload and restore pass) and they're alerady in the cache if flags say they're wanted to be cached

> return the uncached bundle if none of the objects are dummy

Obvs not true as well

Maybe `interm_getAssetInHierarchy_find` name is misleading?